gapic/schema/wrappers.py

```diff
@@ -767,6 +767,29 @@ def http_opt(self) -> Optional[Dict[str, str]]:
         # TODO(yon-mg): enums for http verbs?
         return answer
 
+    @property
+    def path_params(self) -> Sequence[str]:
+        """Return the path parameters found in the http annotation url"""
+        # TODO(yon-mg): fully implement grpc transcoding (currently only handles basic case)
+        if self.http_opt is None:
+            return []
+        pattern = r'\{\w+\}'
+
+        return [x[1:-1] for x in re.findall(pattern, self.http_opt['url'])]
+
+    @property
+    def query_params(self) -> Set[str]:
```

Comment:
Just curious: why can't  `query_params`  and `path_params` return the same type, rather than a `Set` and a `Sequence`?

Reply:
I suppose they could. Just thought it's more appropriate since ordering seems important for `path_params` but not for `query_params`.

Reply:
As discussed. That seems reasonable, but a complication is that repeated fields map to repeated query params. In that case, a sequence may be best. (If order doesn't matter, a multiset might be enough, but we can't assume that order doesn't matter for repeated fields)

Reply:
You are right set is not appropriate even if order doesn't mater but I should mention though that once query param logic is moved out, this won't matter anymore.

Reply:
True!